$NetBSD: patch-regtests_Makefile,v 1.1 2025/10/20 10:20:12 dkazankov Exp $

Run testsuite in WRKDIR instead of /tmp
Fix gprbuild call variables list

--- regtests/Makefile.orig	2024-08-23 18:53:29.000000000 +0300
+++ regtests/Makefile
@@ -20,18 +20,22 @@
 .SILENT:
 endif
 
+TESTSUITE_DIR := $(shell pwd)
+
 -include ../makefile.setup
 
 build:
 	$(GPRBUILD) -q -p -XLIBRARY_TYPE=$(DEFAULT_LIBRARY_TYPE) \
-		$(foreach v,PRJ_TARGET PRJ_BUILD TARGET VERSION,"-X$(v)=$($(v))") \
+		$(foreach v,PRJ_TARGET PRJ_BUILD TARGET TP_XMLADA VERSION,"-X$(v)=$($(v))") \
 		--subdirs=$(SDIR)/$(DEFAULT_LIBRARY_TYPE) -Pregtests
 
 test: build
 	ADA_PROJECT_PATH="$(abspath $(CURDIR)/..)" \
 	PATH=.:${PATH} \
+	LIBRARY_TYPE=$(DEFAULT_LIBRARY_TYPE) \
 	$(foreach v,PRJ_TARGET PRJ_BUILD TARGET TP_XMLADA,$(v)="$($(v))") \
-	./testsuite.py --jobs=4
+	./testsuite.py --jobs=4 -t $(TESTSUITE_DIR)/temp -d $(TESTSUITE_DIR)/dest -o $(TESTSUITE_DIR)/out \
+	    --no-random-temp-subdir --cleanup-mode all --show-error-output
 
 clean:
 	-$(GPRCLEAN) -XLIBRARY_TYPE=$(DEFAULT_LIBRARY_TYPE) \
